### DISCUSSION TITLE
ERROR: DIA-NN tried but failed to load the following files

### DISCUSSION THREAD

**Discussion by witszymanski** [vdemichev/DiaNN#184] [General]
Hi all,
I came across few papers recently and wanted to test out the DIA-NN. I installed MS FIle Reader 3.0 and DIA-NN and... I cannot make it run. It doesn't matter if I try to run DDA .raw files to generate library or DIA files for direct-DIA. I kind of left the default settings for now. The problem is that the DIA-NN cannot load files. 

Anyone can help here? I guess it's something very simple that I miss. Do I need to run is as an admin or something? 
![image](https://user-images.githubusercontent.com/15866068/133080465-a6dde578-af97-4e15-bdb8-7ffe9af6397c.png)

> C:\DIA-NN\1.8\diann.exe --f "D:\DIA-NN\DDA\srRNA-1-1.raw
> " --lib "" --threads 8 --verbose 1 --out "D:\DIA-NN\DDA\Lib\report.tsv" --qvalue 0.01 --matrices  --out-lib "D:\DIA-NN\DDA\Lib\report-lib.tsv" --gen-spec-lib --predictor --fasta "D:\DIA-NN\EcoliK12_fwd.fasta" --fasta-search --min-fr-mz 200 --max-fr-mz 1800 --met-excision --cut K*,R* --missed-cleavages 2 --min-pep-len 7 --max-pep-len 30 --min-pr-mz 300 --max-pr-mz 1800 --min-pr-charge 2 --max-pr-charge 4 --unimod4 --var-mods 1 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --monitor-mod UniMod:1 --reanalyse --smart-profiling 
> DIA-NN 1.8 (Data-Independent Acquisition by Neural Networks)
> Compiled on Jun 28 2021 14:55:31
> Current date and time: Mon Sep 13 14:02:20 2021
> CPU: GenuineIntel Intel(R) Xeon(R) CPU E5-2637 v3 @ 3.50GHz
> SIMD instructions: AVX AVX2 FMA SSE4.1 SSE4.2 
> Logical CPU cores: 16
> Thread number set to 8
> Output will be filtered at 0.01 FDR
> Precursor/protein x samples expression level matrices will be saved along with the main report
> A spectral library will be generated
> Deep learning will be used to generate a new in silico spectral library from peptides provided
> Library-free search enabled
> Min fragment m/z set to 200
> Max fragment m/z set to 1800
> N-terminal methionine excision enabled
> In silico digest will involve cuts at K*,R*
> Maximum number of missed cleavages set to 2
> Min peptide length set to 7
> Max peptide length set to 30
> Min precursor m/z set to 300
> Max precursor m/z set to 1800
> Min precursor charge set to 2
> Max precursor charge set to 4
> Cysteine carbamidomethylation enabled as a fixed modification
> Maximum number of variable modifications set to 1
> Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
> Modification UniMod:1 with mass delta 42.0106 at *n will be considered as variable
> A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step
> When generating a spectral library, in silico predicted spectra will be retained if deemed more reliable than experimental ones
> DIA-NN will optimise the mass accuracy automatically using the first run in the experiment. This is useful primarily for quick initial analyses, when it is not yet known which mass accuracy setting works best for a particular acquisition scheme.
> The following variable modifications will be scored: UniMod:1 
> WARNING: MBR turned off, two or more raw files are required
> 
> 1 files will be processed
> [0:00] Loading FASTA D:\DIA-NN\EcoliK12_fwd.fasta
> [0:01] Processing FASTA
> [0:04] Assembling elution groups
> [0:06] 1011644 precursors generated
> [0:06] Gene names missing for some isoforms
> [0:06] Library contains 4671 proteins, and 4539 genes
> [0:06] Encoding peptides for spectra and RTs prediction
> [0:09] Predicting spectra and IMs
> [7:22] Predicting RTs
> [8:16] Decoding predicted spectra and IMs
> [8:21] Decoding RTs
> [8:22] Saving the library to D:\DIA-NN\DDA\Lib\report-lib.predicted.speclib
> [8:25] Initialising library
> 
> [8:25] File #1/1
> [8:25] Loading run D:\DIA-NN\DDA\srRNA-1-1.raw
> No MS2 spectra: aborting
> ERROR: cannot load the file, skipping
> [8:25] 0 library precursors are potentially detectable
> [8:25] Processing...
> [8:25] Removing low confidence identifications
> [8:25] Searching PTM decoys
> [8:25] Removing interfering precursors
> [8:25] Too few confident identifications, neural networks will not be used
> [8:25] Number of IDs at 0.01 FDR: 0
> [8:25] Calculating protein q-values
> [8:25] Number of genes identified at 1% FDR: 0 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only)
> [8:25] Quantification
> [8:26] Quantification information saved to D:\DIA-NN\DDA\srRNA-1-1.raw.quant.
> 
> ERROR: DIA-NN tried but failed to load the following files: D:\DIA-NN\DDA\srRNA-1-1.raw
> [8:26] Cross-run analysis
> [8:26] Reading quantification information: 1 files
> [8:26] Quantifying peptides
> [8:26] Assembling protein groups
> [8:27] Quantifying proteins
> [8:27] Calculating q-values for protein and gene groups
> [8:27] Calculating global q-values for protein and gene groups
> [8:27] Writing report
> [8:27] Report saved to D:\DIA-NN\DDA\Lib\report.tsv.
> [8:27] Saving precursor levels matrix
> [8:27] Precursor levels matrix (1% precursor and protein group FDR) saved to D:\DIA-NN\DDA\Lib\report.pr_matrix.tsv.
> [8:27] Saving protein group levels matrix
> [8:27] Saving gene group levels matrix
> [8:27] Saving unique genes levels matrix
> [8:27] Stats report saved to D:\DIA-NN\DDA\Lib\report.stats.tsv
> [8:27] Generating spectral library:
> [8:27] Reading quantification information: 1 files
> [8:27] Assembling protein groups
> [8:27] 0 precursors passing the FDR threshold are to be extracted
> [8:27] Saving spectral library to D:\DIA-NN\DDA\Lib\report-lib.tsv
> [8:27] 0 precursors saved
> [8:27] Loading the generated library and saving it in the .speclib format
> [8:27] Loading spectral library D:\DIA-NN\DDA\Lib\report-lib.tsv
> [8:27] Spectral library loaded: 0 protein isoforms, 0 protein groups and 0 precursors in 1 elution groups.
> [8:27] Loading protein annotations from FASTA D:\DIA-NN\EcoliK12_fwd.fasta
> [8:28] Library contains 0 proteins, and 0 genes
> [8:28] Saving the library to D:\DIA-NN\DDA\Lib\report-lib.tsv.speclib
> [8:28] Log saved to D:\DIA-NN\DDA\Lib\report.log.txt
> Finished
> 
> 
> DIA-NN exited
> DIA-NN-plotter.exe "D:\DIA-NN\DDA\Lib\report.stats.tsv" "D:\DIA-NN\DDA\Lib\report.tsv" "D:\DIA-NN\DDA\Lib\report.pdf"
> PDF report will be generated in the background
>

**Comment by vdemichev** [comment]
Hi Witold, 

DDA files are definitely not going to work, with rare exceptions. DIA-NN is meant to be used for the analysis of DIA data.
You can try some publicly available DIA files, they should work. If DIA-NN seems to have a problem with some .raw DIA files you've acquired, please (i) make sure these are not MSX-DIA and if not (ii) run DIA-NN from the command line like this: diann.exe --f [absolute path to the .raw file] --convert --threads 8. Please then share the output displayed by (ii). If not clues of what's going on there too, then please share one of the .raw files it cannot load (the smallest one).

Best,
Vadim

**Comment by vdemichev** [comment]
Yes, you might want to run at as admin. Also, did you install the MSFileReader using the link provided? If not, need to install that specific version, running the installer as admin.

**Reply by witszymanski** [reply]
Damn... That was the case. I did use the MS File Reader 3.0 but SP2 and not SP3. It runs now. 

It's awesome that you guys answer so fast! This is what brings people to the software as much as the quality of the data generated. I'll test DIA-NN now and who know, might be our to-go pipeline soon. 

Greets
Witek

**Reply by vdemichev** [reply]
Thank you. Enjoy, feedback is welcome!

**Reply by GulMuneerBaloch** [reply]
I have tried to search my 3 DIA data files (Thermo data, less complex data) with DIA-NN, but it never completes the analysis. It takes forever and does not finish the search. I searched for more information about DIA-NN and found that MS File Reader is not installed in my system but MS File Reader is not available for free. What to do in this case? Is there any suggestion?
My analysis stuck at "Processing..." for the last 8 hours. 
Thank you in advance.

**Reply by vdemichev** [reply]
Can you please share the logs at Log level = 3? MSFileReader can be downloaded by the link https://thermo.flexnetoperations.com/control/thmo/login?nextURL=%2Fcontrol%2Fthmo%2Fdownload%3Felement%3D6306677. It does require registration, but it's free.

**Reply by GulMuneerBaloch** [reply]
Thank you for your response. I realized MSFileReader was already installed. Please find Log information below. More than 10 hours have been passed now and it's still processing. I don't know what's the problem. Please help me.  

diann.exe --f "C:\Users\XXX_D_DIA_R1.raw
" --f "C:\Users\XXX_D_DIA_R2.raw
" --f "C:\Users\XXX_D_DIA_R3.raw
" --f "C:\Users\XXX_N_DIA_R1.raw
" --f "C:\Users\XXX_N_DIA_R2.raw
" --f "C:\Users\XXX_N_DIA_R3.raw
" --lib "" --threads 3 --verbose 1 --out "C:\DIA-NN\1.8\report.tsv" --qvalue 0.01 --matrices  --fasta "E:\Fasta\1line_sp_mouse_20180529 + iRT.fasta" --fasta-search --min-fr-mz 200 --max-fr-mz 1800 --cut K*,R* --missed-cleavages 2 --min-pep-len 7 --max-pep-len 30 --min-pr-mz 300 --max-pr-mz 1800 --min-pr-charge 2 --max-pr-charge 4 --unimod4 --var-mods 3 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,*n --monitor-mod UniMod:1 --var-mod UniMod:21,79.966331,STY --monitor-mod UniMod:21 --reanalyse --smart-profiling --no-norm 
DIA-NN 1.8 (Data-Independent Acquisition by Neural Networks)
Compiled on Jun 28 2021 14:55:31
Current date and time: Tue Jan  4 11:36:37 2022
CPU: GenuineIntel Intel(R) Core(TM) i7-7820X CPU @ 3.60GHz
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2 
Logical CPU cores: 16
Thread number set to 3
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
Library-free search enabled
Min fragment m/z set to 200
Max fragment m/z set to 1800
In silico digest will involve cuts at K*,R*
Maximum number of missed cleavages set to 2
Min peptide length set to 7
Max peptide length set to 30
Min precursor m/z set to 300
Max precursor m/z set to 1800
Min precursor charge set to 2
Max precursor charge set to 4
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 3
Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
Modification UniMod:1 with mass delta 42.0106 at *n will be considered as variable
Modification UniMod:21 with mass delta 79.9663 at STY will be considered as variable
A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step
When generating a spectral library, in silico predicted spectra will be retained if deemed more reliable than experimental ones
Normalisation disabled
DIA-NN will optimise the mass accuracy automatically using the first run in the experiment. This is useful primarily for quick initial analyses, when it is not yet known which mass accuracy setting works best for a particular acquisition scheme.
The following variable modifications will be scored: UniMod:1 UniMod:21 
WARNING: it's strongly recommended to use deep learning spectra/RTs prediction. If impossible because some unsupported modifications need to be searched, consider using either (i) the --strip-unknown-mods command or (ii) a 'training library' specified with the --learn-lib command. A training library might facilitate about 1.5x more IDs, deep learning - about 2x-3x more IDs.

6 files will be processed
[0:00] Loading FASTA E:\Fasta\1line_sp_mouse_20180529 + iRT.fasta
[0:21] Processing FASTA
[9:12] Assembling elution groups
[20:29] 79954390 precursors generated
[20:31] Protein names missing for some isoforms
[20:31] Gene names missing for some isoforms
[20:31] Library contains 16958 proteins, and 16530 genes
[27:11] Initialising library

[32:01] First pass: generating a spectral library from DIA data
[32:01] File #1/6
[32:01] Loading run C:\Users\XXX_D_DIA_R1.raw
[37:43] 50277605 library precursors are potentially detectable
[41:20] Processing...
[226:23] RT window set to 52.4442
[226:23] Peak width: 3.832
[226:23] Scan window radius set to 8
[226:28] Recommended MS1 mass accuracy setting: 3.72359 ppm

**Reply by GulMuneerBaloch** [reply]
> Can you please share the logs at Log level = 3? MSFileReader can be downloaded by the link https://thermo.flexnetoperations.com/control/thmo/login?nextURL=%2Fcontrol%2Fthmo%2Fdownload%3Felement%3D6306677. It does require registration, but it's free.
diann.exe --f "C:\Users\XXX_D_DIA_R1.raw
" --f "C:\Users\XXX_D_DIA_R2.raw
" --f "C:\Users\XXX_D_DIA_R3.raw
" --f "C:\Users\XXX_N_DIA_R1.raw
" --f "C:\Users\XXX_N_DIA_R2.raw
" --f "C:\Users\XXX_N_DIA_R3.raw
" --lib "" --threads 3 --verbose 1 --out "C:\DIA-NN\1.8\report.tsv" --qvalue 0.01 --matrices --fasta "E:\Fasta\1line_sp_mouse_20180529 + iRT.fasta" --fasta-search --min-fr-mz 200 --max-fr-mz 1800 --cut K*,R* --missed-cleavages 2 --min-pep-len 7 --max-pep-len 30 --min-pr-mz 300 --max-pr-mz 1800 --min-pr-charge 2 --max-pr-charge 4 --unimod4 --var-mods 3 --var-mod UniMod:35,15.994915,M --var-mod UniMod:1,42.010565,n --monitor-mod UniMod:1 --var-mod UniMod:21,79.966331,STY --monitor-mod UniMod:21 --reanalyse --smart-profiling --no-norm
DIA-NN 1.8 (Data-Independent Acquisition by Neural Networks)
Compiled on Jun 28 2021 14:55:31
Current date and time: Tue Jan 4 11:36:37 2022
CPU: GenuineIntel Intel(R) Core(TM) i7-7820X CPU @ 3.60GHz
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2
Logical CPU cores: 16
Thread number set to 3
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
Library-free search enabled
Min fragment m/z set to 200
Max fragment m/z set to 1800
In silico digest will involve cuts at K,R*
Maximum number of missed cleavages set to 2
Min peptide length set to 7
Max peptide length set to 30
Min precursor m/z set to 300
Max precursor m/z set to 1800
Min precursor charge set to 2
Max precursor charge set to 4
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 3
Modification UniMod:35 with mass delta 15.9949 at M will be considered as variable
Modification UniMod:1 with mass delta 42.0106 at *n will be considered as variable
Modification UniMod:21 with mass delta 79.9663 at STY will be considered as variable
A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step
When generating a spectral library, in silico predicted spectra will be retained if deemed more reliable than experimental ones
Normalisation disabled
DIA-NN will optimise the mass accuracy automatically using the first run in the experiment. This is useful primarily for quick initial analyses, when it is not yet known which mass accuracy setting works best for a particular acquisition scheme.
The following variable modifications will be scored: UniMod:1 UniMod:21
WARNING: it's strongly recommended to use deep learning spectra/RTs prediction. If impossible because some unsupported modifications need to be searched, consider using either (i) the --strip-unknown-mods command or (ii) a 'training library' specified with the --learn-lib command. A training library might facilitate about 1.5x more IDs, deep learning - about 2x-3x more IDs.

6 files will be processed
[0:00] Loading FASTA E:\Fasta\1line_sp_mouse_20180529 + iRT.fasta
[0:21] Processing FASTA
[9:12] Assembling elution groups
[20:29] 79954390 precursors generated
[20:31] Protein names missing for some isoforms
[20:31] Gene names missing for some isoforms
[20:31] Library contains 16958 proteins, and 16530 genes
[27:11] Initialising library

[32:01] First pass: generating a spectral library from DIA data
[32:01] File #1/6
[32:01] Loading run C:\Users\XXX_D_DIA_R1.raw
[37:43] 50277605 library precursors are potentially detectable
[41:20] Processing...
[226:23] RT window set to 52.4442
[226:23] Peak width: 3.832
[226:23] Scan window radius set to 8
[226:28] Recommended MS1 mass accuracy setting: 3.72359 ppm

Could you please further guide me about this?

**Reply by vdemichev** [reply]
DIA-NN GUI automatically enables deep learning-based prediction of spectra, RTs and IMs when a FASTA digest is selected. Please keep this on, it's very important. The best way is actually generate an in silico predicted spectral library first (run FASTA digest & deep learning prediction withou specifying any raw data - it will produce a file with the name that ends with '.predicted.speclib'. In the log above deep learning was not enabled, and that's why DIA-NN struggles to identify anything. But in general searching with such a huge search space is not the best idea for DIA. Better to enable only 1 missed cleavage and remove methionine oxidation.

Best,
Vadim
